utils/src/lib.rs

```diff
@@ -239,8 +289,13 @@ pub fn recurse_link<'a>(
             }
 
             // calculate the address for linking this dependency
+            //
+            // NOTE: We add 1 to the nonce, because if we reached this codepath, this contract has a
+            // dependency. At the very least, that one contract will be deployed *prior*
+            // to this one, but wont appear in our deployment vector. It probably should
+            // here, but it doesn't.
```

Comment:
what do you think can go wrong here wrt dependencies?

Reply:
I tested this manually pretty extensively and basically we are doing what we do now, but in a slightly better way, where before there was a footgun when doing the linking. I didn't understand why I needed nonce to be 1 (i.e. in the `link` function) before, but now i do - its because if there is a dependency, fundamentally there is going to be some root contract that has no dependency that will be first. As far as what can go wrong, im not entirely sure, as I said, this is basically what we currently do just in a less dumb way